libpromises/evalfunction.c

```diff
@@ -2202,7 +2212,7 @@ static FnCallResult FnCallGetIndicesClassic(EvalContext *ctx, ARG_UNUSED const P
 
 static FnCallResult FnCallGetIndices(EvalContext *ctx, ARG_UNUSED const Policy *policy, const FnCall *fp, const Rlist *finalargs)
 {
-    const char *name_str = RlistScalarValue(finalargs);
+    const char *name_str = RlistScalarValueSafe(finalargs);
```

Comment:
I've tried to use `RlistScalarValueSafe()` only where we use the `name_str` later. In this specific case, it would complicate the code because of the requirement to call the "classic" way to get indices. I think this is the only place where this happens, so we can adjust it here if you want.

My ultimate goal was to change as little as possible in the code, so if you prefer a longer code flow that explicitly branches on the rlist item type, that's fine, I'll make the change, but it will be a **lot** of code. At that point we could even look at pre-evaluating the data references around function calls, since the boilerplate is very repetitive. But I would prefer to do all that in a cleanup PR.